### ISSUE TITLE
(OS X Mojave) Cinder Audio Input is broken :.(

### ISSUE THREAD

**Issue by rezaali** [cinder/Cinder#2048]
After cloning a clean version of cinder and running the InputAnalyzer sample (on macOS Mojave 10.41.1 Beta) I'm getting nothing (nothing is being visualized). Maybe its the beta OS, but maybe not. Just want to flag this. I <3 Cinder, hope it keeps getting better! Thanks ya'll.

**Comment by notlion** [comment]
Just tested and confirmed this. Also on 10.14.1 (non-beta). Both input from the built-in mic and my external USB audio interface do not work ☹️

**Comment by rezaali** [comment]
I was able to resolve my issue by adding entitlements (enable microphone input) and compiling on macOS 10.14 with Xcode 9.4... and then airdropping the app to my other laptop (macOS 10.14.1).... compiling and running is still not working on 10.14.1 beta... but I'm guessing this is an OS issue, not so much Cinder Audio... Thank you all for chiming in!

**Comment by trevorknight** [comment]
I was able to make the InputAnalyzer sample work using Mojave (10.14.6) with XCode 10.1 (10B61) by adding  the following to the Info.plist:
```
	<key>NSMicrophoneUsageDescription</key>
	<string>This app requires the microphone</string>
	<key>com.apple.security.device.audio-input</key>
	<true/>
```

**Comment by maxmellen** [comment]
Glad to have found this thread!

After digging for a while, it seems only the `NSMicrophoneUsageDescription` key is necessary:
```xml
	<key>NSMicrophoneUsageDescription</key>
	<string>This app requires microphone access.</string>
```

This can also be added in Xcode by opening the `Info.plist` file, adding a new entry and selecting "Privacy - Microphone Usage Description" as the key name from the dropdown menu, then adding the desired message to show the user as the value:
<img width="1072" alt="Screen Shot 2019-12-19 at 16 12 44" src="https://user-images.githubusercontent.com/17215508/71184765-86bb5580-227a-11ea-8f19-05293602c8ab.png">

Here's the Apple Developer Docs page on the topic: https://developer.apple.com/documentation/avfoundation/cameras_and_media_capture/requesting_authorization_for_media_capture_on_macos

**Comment by rezaali** [comment]
thanks @maxmellen, closing this issue.

**Comment by maxmellen** [comment]
@rezaali Would it be worth adding it to the docs somewhere? Maybe in the platform notes of the "Audio in Cinder" guide. If so I'd be happy to make a PR

**Comment by maxmellen** [comment]
Maybe this is a better question for @richardeakin

**Comment by richardeakin** [comment]
Sure, would be happy to see a section added to the audio guide. There's a section called "Platform Specific Notes", source for that is [here](https://github.com/cinder/Cinder/blob/master/docs/htmlsrc/guides/audio/index.html#L581).

Also would appreciate any updates to the audio samples for the latest xcode / OS X / iOS versions. I haven't personally run them or any other code on OS X / xcode in ages..

**Comment by richardeakin** [comment]
I'm re-opening this issue since the InputAnalyzer sample is currently broken on OS X Mohave. Would someone with that OS installed be able to update the sample and submit a PR? We have a release around the corner and probably should address this first.

**Comment by andrewfb** [comment]
Thanks @maxmellen for tracking this fix down. Addressed by f287eaf
